gptcache/manager/scalar_data/sql_storage.py

```diff
@@ -215,6 +246,50 @@ def count(self, state: int = 0, is_all: bool = False):
                 .scalar()
             )
 
+    def add_session(self, question_id, session_id, session_question):
+        with self.Session() as session:
+            session_data = self._session_table(
+                question_id=question_id,
+                session_id=session_id,
+                session_question=session_question,
+            )
+            session.add(session_data)
+            session.commit()
+
+    def delete_session(self, keys):
+        with self.Session() as session:
+            session.query(self._session_table).filter(self._session_table.id.in_(keys)).delete()
+            session.commit()
+
+    def get_sessions_by_key(self, key):
+        with self.Session() as session:
+            res = (
+                session.query(self._session_table.session_id, self._session_table.session_question)
+                       .filter(self._session_table.question_id == key).all()
+            )
+        return [r[0] for r in res], [r[1] for r in res]
+
+    def get_keys_by_session(self, session_id):
+        with self.Session() as session:
+            res = (
+                session.query(self._session_table.id).filter(self._session_table.session_id == session_id).all()
+            )
+        return [r[0] for r in res]
+
+    def get_all_sessions(self):
+        with self.Session() as session:
+            res = (
+                session.query(self._session_table.session_id).all()
+            )
+        return set(r[0] for r in res)
+
+    def list_sessions(self, session_id=None, key=None):
+        if session_id:
+            return self.get_keys_by_session(session_id)
+        if key:
+            return self.get_sessions_by_key(key)
+        return self.get_all_sessions()
```

Comment:
`get_sessions_by_key` return two values, other method return a value
maybe it's more suitable, like:
```
with self.Session() as session:
            query=session.query(self._session_table.id)
            if session_id:
                   query=query.filter(self._session_table.session_id == session_id)
            if key:
                   query=query.filter(self._session_table.question_id == key)
            res = query.all()
            return [r[0] for r in res]
```